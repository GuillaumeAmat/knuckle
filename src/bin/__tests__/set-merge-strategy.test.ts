import { binScriptPath } from '../../helpers/tests/binScriptPath';
import * as cmd from '../../helpers/tests/cmd';
import { getKnuckleConfig } from '../../helpers/tests/getKnuckleConfig';
import { setupPristineTestFolder } from '../../helpers/tests/setupPristineTestFolder';

describe('SET-MERGE-STRATEGY', () => {
  setupPristineTestFolder();

  it('should fail if the given tool does not exists', async done => {
    let hasThrown = false;

    try {
      await cmd.run('node', [binScriptPath, 'set-merge-strategy', 'unknown-tool', 'deep']);
    } catch (err) {
      hasThrown = true;
      expect(err).toMatchSnapshot();
    }

    expect(hasThrown).toBe(true);

    done();
  });

  it('should fail if the given strategy does not exists', async done => {
    let hasThrown = false;

    try {
      await cmd.run('node', [binScriptPath, 'set-merge-strategy', 'prettier', 'unknown-strategy']);
    } catch (err) {
      hasThrown = true;
      expect(err).toMatchSnapshot();
    }

    expect(hasThrown).toBe(true);

    done();
  });

  it('should set merge strategies', async done => {
    await cmd.run('node', [binScriptPath, 'set-merge-strategy', 'default', 'replace']);
    await cmd.run('node', [binScriptPath, 'set-merge-strategy', 'prettier', 'spread']);
    await cmd.run('node', [binScriptPath, 'set-merge-strategy', 'eslint', 'deep']);

    const config = getKnuckleConfig()?.config;

    expect(config.mergeStrategies.default).toBe('replace');
    expect(config.mergeStrategies.prettier).toBe('spread');
    expect(config.mergeStrategies.eslint).toBe('deep');

    done();
  });
});
