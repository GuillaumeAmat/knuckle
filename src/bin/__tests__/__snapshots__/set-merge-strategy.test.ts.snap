// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SET-MERGE-STRATEGY should fail if the given strategy does not exists 1`] = `
"
"
`;

exports[`SET-MERGE-STRATEGY should fail if the given tool does not exists 1`] = `
"
"
`;
